cloud-annotations/src/main/java/org/incendo/cloud/annotations/AnnotationParser.java

```diff
@@ -708,19 +708,43 @@ public void defaultValueRegistry(final @NonNull DefaultValueRegistry<C> defaultV
      * @param <T>      Type of the instance
      * @return Collection of parsed commands
      */
-    @SuppressWarnings({"unchecked", "rawtypes"})
     public <T> @NonNull Collection<org.incendo.cloud.@NonNull Command<C>> parse(final @NonNull T instance) {
-        this.parseDefaultValues(instance);
-        this.parseSuggestions(instance);
-        this.parseParsers(instance);
-        this.parseExceptionHandlers(instance);
-
-        final Collection<CommandDescriptor> commandDescriptors = this.commandExtractor.extractCommands(instance);
-        final Collection<org.incendo.cloud.Command<C>> commands = this.construct(instance, commandDescriptors);
-        for (final org.incendo.cloud.Command<C> command : commands) {
-            ((CommandManager) this.manager).command(command);
+        return this.parse(new Object[]{instance});
+    }
+
+    /**
+     * Scan some classes instances of {@link Command} annotations and attempt to
+     * compile them into {@link org.incendo.cloud.Command} instances.
+     *
+     * @param instances Instances to scan
+     * @param <T>      Type of the instance
+     * @return Collection of parsed commands
+     */
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    public <T> @NonNull Collection<org.incendo.cloud.@NonNull Command<C>> parse(final @NonNull T... instances) {
+        for (T instance: instances) {
```

Comment:
The checkstyle not flagging this is concerning... `:` as part of enhanced for needs a single space on either side (`Type t : iterable`). `instance` should also be final.

Reply:
Fixed